n2v/methods/zmp.py

```diff
@@ -248,22 +252,34 @@ def zmp_scf(self,
                 if SCF_ITER == maxiter - 1:
                     raise ValueError("Maximum Number of SCF cycles reached. Try different settings.")
 
-            density_current = self.on_grid_density(grid=None, Da=Da, Db=Db, vpot=self.vpot)
-            grid_diff = np.max(np.abs(D0 - density_current))
-            if np.abs(grid_diff_old) < np.abs(grid_diff):
-                print("\nZMP halted. Density Difference is unable to be reduced")
-                break
-
-            grid_diff_old = grid_diff
-            print(f"SCF Converged for lambda:{int(lam_i):5d}. Max density difference: {grid_diff}")
-
-            self.Da = Da
-            self.Db = Db
+            # Why do we need this? Basis set -> grid is expensive.
+            # density_current = self.on_grid_density(grid=None, Da=Da, Db=Db, vpot=self.vpot)
+            # grid_diff = np.max(np.abs(D0 - density_current))
+            # if np.abs(grid_diff_old) < np.abs(grid_diff):
+            #     print("\nZMP halted. Density Difference is unable to be reduced")
+            #     break
+            #
+            # grid_diff_old = grid_diff
+            print(f"SCF Converged for lambda:{int(lam_i):5d}. Max density difference: {np.linalg.norm(Da - self.nt[0])}")
+
+            #VXC is hartree-like Potential. We remove Fermi_Amaldi Guess.
+            self.proto_density_a += lam_i * (Da - self.nt[0]) * self.mixing
```

Comment:
I know the mixing here is reducing the amount of the addition to the accumulation, but shouldn't we be also removing from self.proto_density_a as well? They're similar ideas, but not the same.

Reply:
Could you say it again? I don't get it.

Reply:
You add a portion of the new potential. But you're not removing anything from the old potential as in (1-mixing) * vc_old + mixing * vc_new
Just trying to understand if your intention was to do a mix, or just to limit how much of the new potential gets added.

Reply:
Yes you are right. This is strictly speaking not a liner mixing, but rather an accumulation coefficient. The thing is linear mixing does not work smoothly with accumulation. I added lines (commented off) for real linear mixing that you can try.